h/services/jwt.py

```diff
@@ -17,7 +17,9 @@ class JWTService:
     LEEWAY = datetime.timedelta(seconds=10)
 
     @classmethod
-    def decode_token(cls, token: str, key_set_url: str) -> dict[str, Any]:
+    def decode_token(
+        cls, token: str, key_set_url: str, algorithms: list[str]
+    ) -> dict[str, Any]:
```

Comment:
This is a generic JWT service but the list of allowed algorithms is context-specific (e.g. specific to decoding OpenID Connect ID tokens) so had to add the algorithms as a required parameter here for the caller to pass in.